[kmac,doc] Correct performance numbers

vogelpi opened the pull request:
It wasn't clear whether the previous throughput numbers in the documentation were actually correct and how they were obtained. This commit updates the numbers according to the following thinking:

It always takes 25 clock cycles to load the data and the padding into the Keccak core (1600 bits, 64 bits per cycle) and then either 24 or 96 clock cycles to perform the 24 unmasked or masked Keccak rounds, respectively. This results in 144 (SHA3-224) or 72 bytes (SHA3-512) being processed every 49 (unmasked) or 121 clock cycles (masked).

rswarbrick reviewed (commented):
I think it would probably be helpful to add an explicit line on how to convert between Mb/s (the performance goals above) and B/cycle numbers, given a clock frequency.

Otherwise a reader won't immediately know whether the numbers match the performance goals.

vogelpi commented:
> I think it would probably be helpful to add an explicit line on how to convert between Mb/s (the performance goals above) and B/cycle numbers, given a clock frequency.
> 
> Otherwise a reader won't immediately know whether the numbers match the performance goals.

Thanks for your feedback @rswarbrick . I've now also added the performance in Mbit/s. It's much higher than the previously stated performance target. To be honest, I don't know where this target figure came from and now removed it.

rswarbrick reviewed (approved):
Looks great, thanks!
